### PR TITLE
Using typeahead to autocomplete teams when creating a namespace

### PR DESCRIPTION
**Pull request by jloehel**
Fixes: #558

Signed-off-by: Jürgen Löhel <jloehel@suse.com>

### CODE REVIEW

File: app/models/team_user.rb [hunk 1]
```diff
@@ -25,6 +25,11 @@ def create_activity!(type, owner, parameters = nil)
     team.create_activity type, owner: owner, recipient: user, parameters: params
   end
 
+  # Returns all team IDs which are manageable by one user
+  def self.get_valid_team_ids(id)
+    where(user_id: id, role: 2).pluck(:team_id)
```
Comment (mssola) [inline app/models/team_user.rb]: What's role 2 ? Is it owner ? If so, then rewrite it using the `owner` scope:

```ruby
owner.where(user_id: id).pluck(:team_id)
```

File: app/models/team.rb [hunk 2]
```diff
@@ -26,4 +26,9 @@ def self.all_non_special
   def member_ids
     team_users.pluck(:user_id)
   end
+
+  # Returns all teams whose name match the query
+  def self.search_from_query(valid_teams, query)
+    where(id: valid_teams).where(hidden: 0).where(arel_table[:name].matches(query))
```
Comment (mssola) [inline app/models/team.rb]: Use the `all_non_special` method for the `where(hidden: 0)` part. Moreover, could you transform the `all_non_special` method into a scope ? (while preserving all its documentation of course).

File: app/controllers/teams_controller.rb [hunk 3]
```diff
@@ -44,7 +44,7 @@ def typeahead
     authorize @team
     @query = params[:query]
     matches = User.search_from_query(@team.member_ids, "#{@query}%").pluck(:username)
-    matches = matches.map { |user| { username: user } }
+    matches = matches.map { |user| { name: user } }
```
Comment (mssola) [inline app/controllers/teams_controller.rb]: How doesn't this break the form for auto-completing user names in the "Add member" form ?

Comment (jloehel) [inline app/controllers/teams_controller.rb]: I rewrote the javascript in teams.coffee. Now I use one function for both forms.

Comment (mssola) [inline app/controllers/teams_controller.rb]: Ah ok, didn't notice this change :)

**Pull request merged**